kate_brownbill_dungon_fighters_v1.0.py

```diff
@@ -0,0 +1,838 @@
+        if result == 1:
+            jester = True
+    elif ans == 1:
+        print("\nWelcome to dungeon fighters!")
+        print("Here you command a party of adventurers who fight various foes.")
+        print("There are 5 moves, FIREBOLT, FORWARDS SWING, MAGIC SWING")
+        print("SHIELD BASH, and MAGIC REFLECT. Every fighter has some")
+        print("combanation of moves. Once the two fighters have picked a move")
+        print("to use, the two fight. Depending on what moves they used, one")
+        print("fighter will gain the advantage in the battle and deal damage.")
+        print("\nEach fighter as 3 stats: HP ATTC DEF. HP is the hit points")
+        print("ATTC is linked directly to damage given by a fighter. DEF absorbs")
+        print("damage given by an oponent, however DEF will not compleatly absorb")
+        print("all the damage.\n")
+    elif ans == 2:
+        menu = False
+    elif ans == 3:
+        jester = True
+        print("Jester's game unlocked")
+    elif ans == 4:
+        jester_game()
+        
+
+#---------------------------UNITTESTING-------------------------
+def test_ai():
+    fighter1 = Fighter([FIREBOLT, MAGIC_SWNG], "fighter1", "a fighter", 1, 1, 1)
+    fighter2 = Fighter([SHIELD_BASH, FRWD_SWNG, MAGIC_SWNG], "fighter2", "a fighter", 1, 1, 1)
+    
+    turnA1 = agressive_ai(SHIELD_BASH, fighter1) #should return magic swing
+    turnA2 = agressive_ai(FIREBOLT, fighter2) #should return sheild bash
+    turnP1 = passive_ai(SHIELD_BASH, fighter1) #should return firebolt
+    turnP2 = passive_ai(SHIELD_BASH, fighter2) #should return sheild bash
+    turnC1 = copycat_ai(FRWD_SWNG, fighter1) #should return magic swing
+    turnC2 = copycat_ai(FRWD_SWNG, fighter2) #should return forward swing
+    
+    equal_assert(turnA1, (1, MAGIC_SWNG, fighter1))
+    equal_assert(turnA2, (1, SHIELD_BASH, fighter2))
+    equal_assert(turnP1, (1, FIREBOLT, fighter1))
+    equal_assert(turnP2, (1, SHIELD_BASH, fighter2))
+    equal_assert(turnC1, (1, MAGIC_SWNG, fighter1))
+    equal_assert(turnC2, (1, FRWD_SWNG, fighter2))
+
+def test_damage():
+    fighter1 = Fighter([FRWD_SWNG], "fighter1", "a fighter", 2, 1, 1)
+    fighter2 = Fighter([FRWD_SWNG], "Fighter2", "A fighter", 2, 1, 0)
+    fighter3 = Fighter([FRWD_SWNG], "Fighter3", "a fighter", 1, 1, 0)
+
+    fighter1.damage(2) #state should be 1, hp should be 1
+    fighter2.damage(2) #state should be 0, hp should be 0
+    fighter3.damage(1) #state should be 0, hp should be 0
+    
+    equal_assert(fighter1.hp, 1)
+    equal_assert(fighter2.hp, 0)
+    equal_assert(fighter3.hp, 0)
+    
+    equal_assert(fighter1.state, 1)
+    equal_assert(fighter2.state, 0)
+    equal_assert(fighter3.state, 0)
+
+def test_restore():
+    fighter1 = Fighter([FRWD_SWNG], "Fighter1", "A fighter", 8, 1, 0)
+    fighter2 = Fighter([FRWD_SWNG], "Fighter1", "A fighter", 2, 1, 0)
+    
+    fighter1.damage(4)
+    fighter2.damage(4)
+    
```

Comment:
Sometimes it's nice to add something like..

```
    fighter1 = Fighter([FRWD_SWNG], "Fighter1", "A fighter", 8, 1, 0)
    equal_assert( fighter1.state, STATE_ALIVE )  # starts with 8hp and alive
    equal_assert( fighter1.hp, 8 )
    fighter1.damage(4)  # do 4 damage, no defense
    equal_assert( fighter1.state, STATE_ALIVE )  # now has 4hp, still alive
    equal_assert( fighter1.hp, 4 )
    fighter1.restore() # restore to original health
    equal_assert( fighter1.state, STATE_ALIVE )
    equal_assert( fighter1.hp, 8 )
```

As a well of telling the reader what happened.  i.e, I made a fighter,  I damaged it for 4,  it started with 8 hp so it's still alive, but now with 4 hp, I restore the fighter, and now it has 8 hp and it's alive.

Like a story, but with asserts.